tools/memap.py

```diff
@@ -616,6 +619,77 @@ def generate_table(self, file_desc):
 
         return output
 
+    def generate_bars(self, file_desc, device_name=None):
+        """ Generates nice looking bars that represent the memory consumption
+
+        Returns: string containing nice looking bars
+        """
+
+        # TODO add tty detection, and width detection probably
+        WIDTH = 72
+        try:
+            # NOTE this only works on linux
+            import sys, fcntl, termios, struct
+            height, width, _, _ = struct.unpack('HHHH',
+                fcntl.ioctl(sys.stdout.fileno(), termios.TIOCGWINSZ,
+                    struct.pack('HHHH', 0, 0, 0, 0)))
+            WIDTH = min(width, WIDTH)
+        except:
+            pass
```

Comment:
Could you assign width to 72 here instead? and make that except not catch C-c and similar with
```python
except Exception:
    WIDTH = 72
```

Reply:
WIDTH is also used on line 636. Will update the second part of your sentence.